src/Analyser/TypeSpecifier.php

```diff
@@ -900,8 +902,19 @@ private function specifyTypesForConstantBinaryExpression(
 				}
 				$argType = $scope->getType($exprNode->getArgs()[0]->value);
 				if ($argType->isArray()->yes()) {
+					if (count($exprNode->getArgs()) === 1) {
+						$isNormalCount = true;
+					} else {
+						$mode = $scope->getType($exprNode->getArgs()[1]->value);
+						$isNormalCount = (new ConstantIntegerType(COUNT_NORMAL))->isSuperTypeOf($mode)->yes();
+
+						if (!$isNormalCount) {
+							$isNormalCount = $argType->getIterableValueType()->isScalar()->yes();
```

Comment:
This is too restrictive. For example objects cannot be counted recursively, not even countables: https://3v4l.org/3GCpt

So you need to ask `isArray` and work with uncertainty - TrinaryLogic. Think about the result when it's definitely not an array, when it can be an array and when it's definitely an array.